### DISCUSSION TITLE
How to filter data through Legend click

### DISCUSSION THREAD

**Discussion by TinsFox** [recharts/recharts#3940] [Q&A] [answered]
When I have many sets of data, can I hide certain data by clicking the Legend? Sometimes I only want to focus on certain data.

Just like the following [example](https://g2.antv.antgroup.com/examples/general/line/#line-aggregated-label)

https://github.com/recharts/recharts/assets/33956589/6cb33739-5f6f-4bf2-86ed-2f03da258d89

Is such a feature supported now? If so, how should I write the code? Thank you all

**Comment by ckifer** [comment, marked as answer]
This should help you out 

https://master--63da8268a0da9970db6992aa.chromatic.com/?path=/story/examples-linechart--hide-on-legend-click

https://github.com/recharts/recharts/blob/master/storybook/stories/Examples/LineChart.stories.tsx#L967

**Reply by TinsFox** [reply]
Thank you very much, this solved my problem!
